pkgs/flox-activation-scripts/default.nix

```diff
@@ -102,18 +93,47 @@ runCommandNoCC "flox-activation-scripts"
       $out/activate.d/generate-fish-startup-commands.bash \
       $out/activate.d/generate-tcsh-startup-commands.bash \
       $out/activate.d/set-prompt.bash \
-      $out/etc/profile.d/* \
-      $build_wrapper/activate \
-      $build_wrapper/activate.d/generate-bash-startup-commands.bash \
-      $build_wrapper/activate.d/generate-fish-startup-commands.bash \
-      $build_wrapper/activate.d/generate-tcsh-startup-commands.bash \
-      $build_wrapper/activate.d/set-prompt.bash \
-      $build_wrapper/etc/profile.d/*
+      $out/activate.d/source-profile-d.bash \
+      $out/etc/profile.d/*
 
     # Finally check the formatting of the scripts with shfmt. Only check
     # $out as the contents of $build_wrapper will be virtually identical.
     cp ${editorconfig} $out/.editorconfig
     # This will only catch extensions and shebangs that `shfmt --find` knows about.
     ${shfmt}/bin/shfmt --diff $out
     rm $out/.editorconfig
+
+    # Remove the wrapper script from the output,
+    # we dont need that in environment interpreters.
+    rm $out/wrapper
+
+    # Next create the (lesser) "build_wrapper" output.
+    # TODO: come up with neater way to master activate script for build_wrapper case.
+
+    mkdir -p $build_wrapper
+    cp ${activation-scripts}/wrapper $build_wrapper/
+
+    # create activate.d directory and copy the required scripts
+    mkdir -p $build_wrapper/activate.d
+    cp ${activation-scripts}/activate.d/source-profile-d.bash $build_wrapper/activate.d/
+    cp ${activation-scripts}/activate.d/trace.bash $build_wrapper/activate.d/
+
+    # copy the etc/profile.d directory
+    cp -R ${activation-scripts}/etc $build_wrapper/
```

Comment:
It was mentioned that we may not need/want all of the profile scripts here in the first place. Though i'm lacking context as for 1. why, 2. how that relates to "TODO"s in the code asking to use profile scripts from the environment instead of bundling them and 3. if that is a necessary change we need to do right now.

Reply:
1. Left another comment on https://github.com/flox/flox/issues/2691
2. and 3. I don't think we need to change it in this PR